File: src/main/java/com/example/cineticket_pro/entities/Usuario.java
package com.example.cineticket_pro.entities;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor

public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    public String nome;
    public int telefone;
    public String email;
    public String senha;
    public String filme; //fk
    public String sessao;//fk
    public String ingresso;//fk
    private EnumStatusUsuario status;
}

File: front/app/(sistema)/types/usuario.ts
export class Usuario{
    constructor(
       public id: number| null,
       public nome:string,
       public telefone:string,
       public email: string,
       public senha: string,
       public status: string,


    ){}
}

export interface UsuarioFormProps{
    usuarioExistente?:Usuario
}